go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -586,6 +541,198 @@ func (c *Controller) generateFRRConfigurations(ra *ratypes.RouteAdvertisements)
+			subnets := selectedNetworks.networkSubnets[network]
+			networkVRFsSubnets[networkVRF] = subnets
+		}
+	} else {
+		networkVRFsSubnets[targetVRF] = selectedNetworks.subnets
+	}
+	return networkVRFsSubnets
+}
+
+// checkSubnetSliceOverlaps checks if subnet within a list of subnets overlap with each other
+func checkSubnetSliceOverlaps(subnets []string) ([]string, error) {
+	for i, subnet1 := range subnets {
+		for j, subnet2 := range subnets {
+			if i >= j {
+				continue
+			}
+			overlap, err := subnetsOverlap(subnet1, subnet2)
+			if err != nil {
+				return nil, fmt.Errorf("failed to check subnet overlap between %s and %s: %w", subnet1, subnet2, err)
+			}
+			if overlap {
+				return []string{subnet1, subnet2}, nil
+			}
+		}
+	}
+	return []string{}, nil
+}
+
+// checkCrossRAOverlaps checks if subnets from current RA overlap with subnets from other RAs in the same VRF
+func (c *Controller) checkCrossRAOverlaps(ra *ratypes.RouteAdvertisements, targetVRF string, currentRASubnets []string) error {
+	allRAs, err := c.raLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, otherRA := range allRAs {
+		if otherRA.Name == ra.Name {
+			continue
+		}
+
+		nads, err := c.getSelectedNADs(otherRA.Spec.NetworkSelectors)
+		if err != nil {
+			return err
+		}
+		if len(nads) == 0 {
+			return fmt.Errorf("%w: failed checking cross RA overlap with RouteAdvertisement %q: no networks selected", errPending, ra.Name)
+		}
+		otherRASelectedNetworks, err := c.getSelectedNetworkInfo(nads, sets.New(otherRA.Spec.Advertisements...))
+		if err != nil {
+			// If we can't get networks from another RA, log but don't fail
+			// This could happen if the other RA has configuration issues
+			klog.V(4).Infof("Failed to get selected networks from RouteAdvertisement %q for cross overlap check with RouteAdvertisement %q: %v", otherRA.Name, ra.Name, err)
+			continue
+		}
+
+		otherVRFSubnets := createNetworkVRFsSubnetsMap(otherRA.Spec.TargetVRF, otherRASelectedNetworks)
+		otherSubnets, hasMatchingVRF := otherVRFSubnets[targetVRF]
+		if !hasMatchingVRF || len(otherSubnets) == 0 {
+			continue
+		}
+
+		// Check for overlaps between current RA subnets and other RA subnets in the same VRF
+		for _, currentSubnet := range currentRASubnets {
+			for _, otherSubnet := range otherSubnets {
+				overlap, err := subnetsOverlap(currentSubnet, otherSubnet)
+				if err != nil {
+					return fmt.Errorf("failed to check subnet overlap between %s and %s: %w", currentSubnet, otherSubnet, err)
+				}
+				if overlap {
+					subnetsByRA := map[string]string{
+						ra.Name:      currentSubnet,
+						otherRA.Name: otherSubnet,
+					}
+					raNames := []string{ra.Name, otherRA.Name}
+					slices.Sort(raNames)
+					conflictErr := fmt.Errorf("%w: overlapping CIDR detected between RouteAdvertisements %q and %q in VRF %q: [%s %s]",
+						errConfig, raNames[0], raNames[1], targetVRF, subnetsByRA[raNames[0]], subnetsByRA[raNames[1]])
```

Comment:
Why do we complicate this up without need ? 

```suggestion
				if !overlap {
				        continue
				 }
				conflictErr := fmt.Errorf("%w: overlapping CIDR detected between RouteAdvertisements %q and %q in VRF %q: [%s %s]",
						errConfig, ra.Name, otherRA.Name, targetVRF, currentSubnet, otherSubnet)
```